Configurable-Tasks

lintangsutawika opened the pull request:
- [ ] Make ConfigurableTasks for all request types
- [ ] Make default configurations for available tasks.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=438) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=438) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=438) it.</sub>

lintangsutawika commented:
@haileyschoelkopf do you have an example task for loglikelihood_rolling that I can port to the yaml version?

Also, I've 

1. updated yamls for lambada, gsm8k, and arc.
2. incorporated the `use_prompt` into the doc_to_text process and `doc_to_decontamination_query`
3. updated how metrics are enumerated
4. removed `stop_sequence` in favor or `delimiter`
5. configurable_task can process `greedy_until` and `loglikelihood` but not yet `loglikelihood_rolling`

haileyschoelkopf commented:
For loglikelihood_rolling I've been using Wikitext! I think it's ~just PerplexityTasks that use that request type

Awesome!!!! Are you still feeling we should separate out MultipleChoice as its own "task.output_type"?

What you've got rn looks great to me. We can maybe revisit stop_sequence in the event we do want to enumerate many stop sequences later

haileyschoelkopf commented:
Thanks for this! Merging so that we're back to working from `big-refactor`. All task types are now supported via configurable form right now.
